src/onnxruntime.cc

```diff
@@ -388,221 +388,219 @@ ModelState::LoadModel(
+  if (instance_group_kind != TRITONSERVER_INSTANCEGROUPKIND_CPU) {
+    // Default GPU execution provider.
+    // Using default values for everything other than device id and cuda
+    // stream
+    OrtCUDAProviderOptions cuda_options;
+    cuda_options.device_id = instance_group_device_id;
+    cuda_options.has_user_compute_stream = stream != nullptr ? 1 : 0;
+    cuda_options.user_compute_stream =
+        stream != nullptr ? (void*)stream : nullptr,
+    cuda_options.default_memory_arena_cfg = nullptr;
 
-      RETURN_IF_ORT_ERROR(ort_api->SessionOptionsAppendExecutionProvider_CUDA(
-          soptions, &cuda_options));
-      LOG_MESSAGE(
-          TRITONSERVER_LOG_VERBOSE,
-          (std::string("CUDA Execution Accelerator is set for '") + Name() +
-           "' on device " + std::to_string(instance_group_device_id))
-              .c_str());
+    {
+      // Parse CUDA EP configurations
+      triton::common::TritonJson::Value params;
+      if (model_config_.Find("parameters", &params)) {
+        int cudnn_conv_algo_search = 0;
+        RETURN_IF_ERROR(TryParseModelStringParameter(
+            params, "cudnn_conv_algo_search", &cudnn_conv_algo_search, 0));
+        cuda_options.cudnn_conv_algo_search =
+            static_cast<OrtCudnnConvAlgoSearch>(cudnn_conv_algo_search);
+
+        RETURN_IF_ERROR(TryParseModelStringParameter(
+            params, "gpu_mem_limit", &cuda_options.gpu_mem_limit,
+            std::numeric_limits<size_t>::max()));
+
+        RETURN_IF_ERROR(TryParseModelStringParameter(
+            params, "arena_extend_strategy",
+            &cuda_options.arena_extend_strategy, 0));
+
+        RETURN_IF_ERROR(TryParseModelStringParameter(
+            params, "do_copy_in_default_stream",
+            &cuda_options.do_copy_in_default_stream, true));
+      }
     }
+
+    RETURN_IF_ORT_ERROR(ort_api->SessionOptionsAppendExecutionProvider_CUDA(
+        soptions, &cuda_options));
+    LOG_MESSAGE(
+        TRITONSERVER_LOG_VERBOSE,
+        (std::string("CUDA Execution Accelerator is set for '") + Name() +
+         "' on device " + std::to_string(instance_group_device_id))
+            .c_str());
+  }
 #endif  // TRITON_ENABLE_GPU
 
-    // CPU execution providers
-    {
-      triton::common::TritonJson::Value optimization;
-      if (model_config_.Find("optimization", &optimization)) {
-        triton::common::TritonJson::Value eas;
-        if (optimization.Find("execution_accelerators", &eas)) {
-          triton::common::TritonJson::Value cpu_eas;
-          if (eas.Find("cpu_execution_accelerator", &cpu_eas)) {
-            for (size_t ea_idx = 0; ea_idx < cpu_eas.ArraySize(); ea_idx++) {
-              triton::common::TritonJson::Value ea;
-              RETURN_IF_ERROR(cpu_eas.IndexAsObject(ea_idx, &ea));
-              std::string name;
-              RETURN_IF_ERROR(ea.MemberAsString("name", &name));
+  // CPU execution providers
+  if (instance_group_kind != TRITONSERVER_INSTANCEGROUPKIND_AUTO) {
```

Comment:
I don't think this condition is required. If the config provides `cpu_execution_accelerator` then only this will get added.
For KIND_AUTO, the config might still provide a CPU accelerator and this condition would render it ineffective.